kclvm/driver/src/lib.rs

```diff
@@ -111,7 +114,38 @@ pub fn lookup_compile_unit(file: &str) -> (Vec<String>, Option<LoadProgramOption
                 Err(_) => return (vec![file.to_string()], None),
             }
         }
-        Err(_) => return (vec![file.to_string()], None),
+        Err(_) => {
+            if load_pkg {
+                let path = Path::new(file);
+                if let Some(ext) = path.extension() {
+                    if ext == KCL_FILE_EXTENSION && path.is_file() {
+                        if let Some(parent) = path.parent() {
+                            if let Ok(entries) = fs::read_dir(parent) {
+                                let mut files = Vec::new();
+                                for entry in entries {
+                                    if let Ok(entry) = entry {
+                                        if let Some(ext) = entry.path().extension() {
+                                            if ext == KCL_FILE_EXTENSION {
+                                                if let Some(file) = entry.path().to_str()
+                                                {
+                                                    files.push(file.to_string());
+                                                }
+                                            }
+                                        }
+                                    }
+                                }
+                                if !files.is_empty() {
+                                    return (files, None);
+                                }
+                            }
+                        }
+                    }
+                }
+                return (vec![file.to_string()], None);
```

Comment:
The if and else branches seem to have returned the same content.

Reply:
It will return early if found some pkg files